nx-dev/data-access-documents/src/lib/blog.api.ts

```diff
@@ -75,6 +80,64 @@ export class BlogApi {
     return sortPosts(allPosts);
   }
 
+  getBlogPosts(): BlogPostDataEntry[] {
+    const files: string[] = readdirSync(this.options.blogRoot);
+    const authors = JSON.parse(
+      readFileSync(join(this.options.blogRoot, 'authors.json'), 'utf8')
+    );
+    const allPosts: BlogPostDataEntry[] = [];
+
+    for (const file of files) {
+      const filePath = join(this.options.blogRoot, file);
+      // filter out directories (e.g. images)
+      if (!filePath.endsWith('.md')) continue;
+
+      const content = readFileSync(filePath, 'utf8');
+      const frontmatter = extractFrontmatter(content);
+      const slug = this.calculateSlug(filePath, frontmatter);
+      const { image, type } = this.determineOgImage(frontmatter.cover_image);
+      const post = {
+        content,
+        title: frontmatter.title ?? null,
+        description: frontmatter.description ?? null,
+        authors: authors.filter((author) =>
+          frontmatter.authors.includes(author.name)
+        ),
+        date: this.calculateDate(file, frontmatter),
+        cover_image: frontmatter.cover_image
+          ? `/documentation${frontmatter.cover_image}` // Match the prefix used by markdown parser
+          : null,
+        tags: frontmatter.tags ?? [],
+        reposts: frontmatter.reposts ?? [],
+        pinned: frontmatter.pinned ?? false,
+        ogImage: image,
+        ogImageType: type,
+        filePath,
+        slug,
+        podcastYoutubeId: frontmatter.podcastYoutubeId,
+        podcastSpotifyId: frontmatter.podcastSpotifyId,
+        iHeartUrl: frontmatter.iHeartUrl,
+        appleUrl: frontmatter.applePodcastsUrl,
+        amazonUrl: frontmatter.amazonUrl,
```

Comment:
this block seems a duplicate as in the `getAllBlogs`. We should refactor it out to not have to do the mappings twice